src/js/components/Calendar/Calendar.js

```diff
@@ -174,48 +185,59 @@ const Calendar = forwardRef(
     // set activeDate when caller changes it, allows us to change
     // it internally too
     const [activeDate, setActiveDate] = useState(
-      dateProp && range ? activeDates.end : activeDates.start,
+      dateProp && typeof dateProp === 'string' && range
+        ? activeDates.end
+        : activeDates.start,
     );
     useEffect(() => {
       if (activeDateProp) setActiveDate(activeDateProp);
     }, [activeDateProp]);
 
-    // set date when caller changes it, allows us to change it internally too
-    const [date, setDate] = useState(dateProp);
-    useEffect(() => setDate(normalizeForTimezone(dateProp)), [dateProp]);
-
-    // set dates when caller changes it, allows us to change it internally too
-    const [dates, setDates] = useState(datesProp);
-    useEffect(() => {
-      // convert all values to UTC
-      if (Array.isArray(datesProp)) {
-        if (Array.isArray(datesProp[0])) {
+    // function that runs inside the useEffect for date and dates
+    const dateCaller = (dateValue, callbackSetState) => {
+      // convert values to UTC based on if date is string or array
+      if (typeof dateValue === 'string') {
+        callbackSetState(normalizeForTimezone(dateValue));
+      } else if (Array.isArray(dateValue)) {
+        if (Array.isArray(dateValue[0])) {
           let from;
           let to;
-          [from, to] = datesProp[0].map(day =>
+          [from, to] = dateValue[0].map(day =>
             day ? new Date(day) : undefined,
           );
-          if (from) from = normalizeForTimezone(from, datesProp[0][0]);
-          if (to) to = normalizeForTimezone(to, datesProp[0][0]);
+          if (from) from = normalizeForTimezone(from, dateValue[0][0]);
+          if (to) to = normalizeForTimezone(to, dateValue[0][0]);
 
-          setDates([[from, to]]);
+          callbackSetState([[from, to]]);
         } else {
-          const datesArray = [];
-          datesProp.forEach(d => {
+          const dateArray = [];
+          dateValue.forEach(d => {
             if (Array.isArray(d)) {
               let from;
               let to;
               [from, to] = d.map(day => new Date(day));
               from = normalizeForTimezone(from, d[0]);
               to = normalizeForTimezone(to, d[0]);
-              datesArray.push([from, to]);
+              dateArray.push([from, to]);
             } else {
-              datesArray.push(normalizeForTimezone(d));
+              dateArray.push(normalizeForTimezone(d));
             }
           });
-          setDates(datesArray);
+          callbackSetState(dateArray);
         }
-      } else setDates(undefined);
+      } else callbackSetState(undefined);
+    };
+
+    // set date when caller changes it, allows us to change it internally too
+    const [date, setDate] = useState(dateProp);
+    useEffect(() => {
+      dateCaller(dateProp, setDate);
```

Comment:
Given that we are always calling the second argument, it might be a little cleaner to do:
`setDate(normalizeDate(dateProp));`

Reply:
Done